extensions/wikia/Insights/InsightsController.class.php

```diff
@@ -56,10 +66,11 @@ private function renderSubpage() {
 		 */
 		if ( $this->model instanceof InsightsPageModel ) {
 			$params = $this->request->getParams();
-			$this->setVal('content', $this->model->getContent( $params ) );
+			$this->setVal( 'content', $this->model->getContent( $params ) );
 			$this->preparePagination();
 			$this->prepareSortingData();
-			$this->setVal('data', $this->model->getViewData() );
+			$this->renderFlagsFiltering();
```

Comment:
Maybe we can use our existing sorting mechanism instead of making dedicated one only for one Insights type.